File: src/c#/Proxy/GercekBasbakan.cs
using System;
namespace DesignPatterns.org.javaturk.dp.basic.pattern.gof.structural.proxy.basbakan
{
    public class GercekBasbakan : IBasbakan
    {

        public void DertDinle(string dert)
        {
            Console.WriteLine("Basbakan: Dinliyorum.");
        }

        public void IsBul(string yakinim)
        {
            Console.WriteLine("Basbakan: Bana boyle isteklerle gelmeyin.");
        }

    }
}

File: src/java/org/javaturk/dp/ch05/proxy/basbakan/GercekBasbakan.java
package org.javaturk.dp.ch05.proxy.basbakan;

public class GercekBasbakan implements Basbakan {

	@Override
	public void dertDinle(String dert) {
		System.out.println("Basbakan: Dinliyorum.");
	}

	@Override
	public void isBul(String yakinim) {
		System.out.println("Basbakan: Bana boyle isteklerle gelmeyin.");
	}
}
